### DISCUSSION TITLE
Note for concepts to be reviewed

### DISCUSSION THREAD

**Discussion by renellis** [TriliumNext/Trilium#4587] [Ideas]
Apologies for the confusing title, I'm trying to archive the following:

While studying or taking notes, I want to be able to "tag" a specific word or sentence so that I can review it later. For instance, I can put all those "concepts" in a separate note called "Review", in a list, where I can then create subnotes that clarify the concept. 

Is something like that possible?

**Comment by rom1dep** [comment]
Yup, I do that all the time, and Trilium is quite flexible (and powerful!) in this area so you might find something else that works better for you. But in short, what I do is that I have "Directory" node containing hierarchies of what other PKM systems would call "objects" or "types": a structure to store instances and templates for "Persons", "Locations", "Companies", …, each having (inheritable) attributes (and some specializations along the way, e.g. a "Colleague" is a note & template under "Person", which inherits the attributes of "Person", and provides more specialized ones, like "Company").

What that gives me is a collection of contextual templates ready to be filled while taking a note. Let's say that I'm writing about "John Doe", a Person, for which I don't have a note yet, I just type `@` and then "John Doe", and Trilium will propose to create a new note and pick the type I need (in this case, a Person).

This new note:
- will become a child of the current note (this provides hierarchical context, in case you care)
- will backlink to the current note (this provides usage context, in case you care)
- will inherit the "Person Template" and all its attributes (this provides consistency with all other "Persons")
- will have a owned attribute `~template=[Person Template]` (so you can find all instances of "Persons" across your entire database, no matter where they are stored in the hierarchy)

What I do generally is that I leave the new "object note" as a child of the current one for a while, get back to it when I find the time, and once I'm happy with its content, move it somewhere under the "Directory" node. What this hierarchy structure for "Directory" allows is re-qualifying objects and keeping them current over time (e.g. if a "Person" at some point becomes a "Colleague", changing its parenting will enable new properties to be populated). I'm not aware of any other system offering that much flexibility.

**Comment by root-hal9000** [comment]
@rom1dep Thanks for your explanation - maybe I am not clear exactly on your solution as I just tried and it's creating a backlink - but is what you are talking about a potential way for getting at the idea for which  I just opened a new discussion #4588 ?

**Reply by rom1dep** [reply]
I replied in #4588 .

>  I am not clear exactly on your solution as I just tried and it's creating a backlink

Did you create some templates like "Person of Interest Template" and "Location of Interest Template" to begin with? You can get some hints from there: https://github.com/zadam/trilium/wiki/Template

If you then add a tag `#reviewed=false` to those templates (or put them under a parent note that has this attribute as inheritable…), every new instance of "Person of Interest" and "Location of Interest" will have this attribute. You can then find such notes from a search, and even store the query+result of it in a note of its own.

Hope this helps.

**Reply by root-hal9000** [reply]
Thanks, the search solution works great! I might even look into finding a way to customize a special search template, but right now that would be me just doing "productive procrastination" :)
